packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -156,17 +158,52 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
             },
           ],
         };
+      } else if (
+        // source exists
+        Boolean(indices[source!]?.mappings?._meta?.migrationMappingPropertyHashes) &&
+        // ...and mappings are unchanged
+        !diffMappings(
+          /* actual */
+          indices[source!].mappings,
+          /* expected */
+          stateP.targetIndexMappings
+        )
+      ) {
+        const targetIndex = source!;
+        const sourceMappings = indices[source!].mappings;
+
+        return {
+          ...stateP,
+          controlState: 'PREPARE_COMPATIBLE_MIGRATION',
+          sourceIndex: Option.none,
+          targetIndex,
+          sourceIndexMappings: sourceMappings,
+          preTransformDocsActions: [
+            // Point the version alias to the source index. This let's other Kibana
+            // instances know that a migration for the current version has is "done"
+            // even though we may be waiting for document transformations to finish.
+            { add: { index: source!, alias: stateP.versionAlias } },
+            ...buildRemoveAliasActions(source!, Object.keys(aliases), [
+              stateP.currentAlias,
+              stateP.versionAlias,
+            ]),
+          ],
+          targetIndexCurrentMappings: sourceMappings,
```

Comment:
NIT Perhaps I didn't use the best naming for this property, but having `target...: source....` looks confusing. Let's see if we can get to a consensus on how to name this one 🤔 

Perhaps we could perform the comparison at this point (which we are already doing on line 165), and use something like `skipUpdateTargetMappings: true` instead, WDYT?